commands/publish_dafsa.py

```diff
@@ -0,0 +1,96 @@
+import requests
+from requests.exceptions import HTTPError
+import tempfile
+import subprocess
+from kinto_http import Client
+
+COMMIT_HASH_URL = "https://api.github.com/repos/publicsuffix/list/commits/master"
+
+LIST_URL = (
+    "https://raw.githubusercontent.com/publicsuffix/list/master/public_suffix_list.dat"
+)
+
+MAKE_DAFSA_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/make_dafsa.py"
+PREPARE_TLDS_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/prepare_tlds.py"
+
+
+SERVER = "https://kinto.dev.mozaws.net/v1"
+CREDENTIALS = ("arpit73", "s3cr3t")  # (username, password)
+
+
+def handle_request_errors(func):
+    def inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+        else:
+            print("Fetching Failed")
+
+    return inner
+
+
+@handle_request_errors
+def get_latest_hash(url):
+    response = requests.get(url)
+    return response.json()["sha"]
+
+
+@handle_request_errors
+def download_resources(*urls, **kwargs):
+    for url in urls:
+        file_location = kwargs["directory"] + "/" + url.split("/")[-1]
+        print(file_location)
+        response = requests.get(url, stream=True)
+        with open(file_location, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:
+                    f.write(chunk)
+                else:
+                    print("Error!!!")
+
+
+# TODO
+def check_last_stored_hash():
+    return True
+
+
+def publish_dafsa(server, credentials, commit_hash):
+
+    bucket_id = "firefox-core-network-dns"
+    collection_id = "public-suffix-list"
+    record_id = "latest-commit-hash"
+
+    client = Client(server_url=SERVER, auth=credentials)
```

Comment:
The shorter way to make a client while creating a bucket and collection at the same time is to use the this code below  
```python
client = Client(server_url=SERVER, auth=CREDENTIALS, bucket=bucket_id, collection=collection_id)
```
However this does not create a bucket or a collection, pretty sure this is a bug in kinto-http.py library.
```python
client = Client(server_url=SERVER, auth=CREDENTIALS)
client.create_bucket(id=bucket_id)
client.create_collection(id=collection_id, bucket=bucket_id)
```
This more verbose way of writing it works, shall could you test it once more, we should file it as a bug then.

Reply:
> However this does not create a bucket or a collection, pretty sure this is a bug in kinto-http.py library.

What makes you think that?

> This more verbose way of writing it works, shall could you test it once more, we should file it as a bug then.

The script does not have to create the bucket and collection. They will be created once by a sysadmin. For your local developments, or when working with the dev server, you can put the bucket and collection creations in a small shell script apart, as suggested here https://remote-settings.readthedocs.io/en/latest/tutorial-dev-server.html

Reply:
> > However this does not create a bucket or a collection, pretty sure this is a bug in kinto-http.py library.
> 
> What makes you think that?

In my testing, this way of creating buckets and collections doesn't work, I'm not saying this just for our use case but 
but as a bug in kinto-http.py

>The script does not have to create the bucket and collection. They will be created once by a sysadmin.

I thought it would be something like this, since this is an updation script, should I assume that a  record already exists and I will only write functions to update records and not create anything new.
The first record can be uploaded even manually as you mentioned.

Reply:
> In my testing, this way of creating buckets and collections doesn't work, I'm not saying this just for our use case but as a bug in kinto-http.py

Hm, but where did you see that this was a way of creating buckets and collections?

> should I assume that a record already exists and I will only write functions to update records and not create anything new?

The script will only be allowed to create/update/delete records, but not buckets and collections (done by admin). 
Also, you can look at https://github.com/Kinto/kinto-http.py#get-or-create

Reply:
> > In my testing, this way of creating buckets and collections doesn't work, I'm not saying this just for our use case but as a bug in kinto-http.py
> 
> Hm, but where did you see that this was a way of creating buckets and collections?

Saw this in the documentation here https://github.com/Kinto/kinto-http.py/blob/master/README.rst#creating-a-client

I confirmed it by looking at the code for Client object here https://github.com/Kinto/kinto-http.py/blob/master/kinto_http/__init__.py#L83-L95

> The script will only be allowed to create/update/delete records, but not buckets and collections (done by admin).

Got it, my script was creating buckets and collections as well 😬

Reply:
> Saw this in the documentation here https://github.com/Kinto/kinto-http.py/blob/master/README.rst#creating-a-client

I see.. But it says «creating a client». Maybe the docs should be rephrased then. Instantiating a client with assigned bucket id or collection id does not create them.

Reply:
I worked on it a bit :) https://github.com/Kinto/kinto-http.py/pull/209

Reply:
Awesome, nice to see the related kinto projects can improve as a result of our project.